### PR TITLE
preload + invoke for electron

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] add preload
- [x] add KB namespace to global
- [x] deprecate most of safe-electron
- [x] switch app.emit to ipcRenderer.invoke

### to KB
- [x] path helpers
- [x] __dirname
- [x] electron.getAppPath
- [x] process helpers

### CODE REVIEW

File: shared/actions/platform-specific/index.desktop.tsx [hunk 1]
```diff
@@ -230,7 +232,7 @@ const sendKBServiceCheck = (state: Container.TypedState, action: ConfigGen.Daemo
     state.config.daemonHandshakeWaiters.size === 0 &&
     state.config.daemonHandshakeFailedReason === ConfigConstants.noKBFSFailReason
   ) {
-    SafeElectron.getApp().emit('keybase' as any, {type: 'requestStartService'})
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.desktop.tsx]: likely means windows wasn't starting the service

Comment (chrisnojima) [reply shared/actions/platform-specific/index.desktop.tsx]: cc: @xgess

Comment (xgess) [reply shared/actions/platform-specific/index.desktop.tsx]: it was. on my windows machine, the gui is definitely starting the service (i think via this mechanism). and actually, i have a PR up to give that a little tweak: https://github.com/keybase/client/pull/22023/files#diff-27d3a4d2bf9aa2500837b33a82f688a0R83

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Pull request merged**